src/components/IntegrationTest/ContextsField.tsx

```diff
@@ -0,0 +1,164 @@
+ * @param {ComponentKind[]} components - Array of components from which new context options will be created.
+ *
+ * @returns {ContextOption[]} - An array combining the base selected contexts and the newly generated component contexts, each having a `selected` field.
+ */
+export const addComponentContexts = (
+  baseSelectedContexts: ContextOption[],
+  selectedContextNames: string[],
+  components: ComponentKind[],
+) => {
+  const selectedContextNameSet = new Set(selectedContextNames); // Faster lookups
+  const componentContexts = components.map((component) => {
+    const componentName = component.metadata.name;
+    return {
+      // if a component context has been previously selected, it should be prefixed with 'component_'
+      // after it's been selected and saved.
+      name: `component_${componentName}`,
+      description: `execute the integration test when component ${componentName} updates`,
+      selected: selectedContextNameSet.has(`component_${componentName}`), // Check if it's selected
+    };
+  });
+
+  return [...baseSelectedContexts, ...componentContexts];
+};
+
+const ContextsField: React.FC<IntegrationTestContextProps> = ({ heading, fieldName }) => {
+  const { namespace } = useWorkspaceInfo();
+  const { appName } = useParams();
+  const [components, componentsLoaded] = useComponents(namespace, appName);
+
+  const [, { value: contexts }] = useField(fieldName);
+  const selectedContextNames = (contexts ?? []).map((c: ContextOption) => c.name);
+  const selectedContexts = React.useMemo(() => {
+    const initialSelectedContexts = mapContextsWithSelection(selectedContextNames, contextOptions);
+
+    // If we have components and they are loaded, add to context option list.
+    // Else, return the base context list.
+    return componentsLoaded && components
+      ? addComponentContexts(initialSelectedContexts, selectedContextNames, components)
+      : initialSelectedContexts;
+  }, [componentsLoaded, components, selectedContextNames]);
+
+  /**
+   * React callback that is used to select or deselect a context option using Formik FieldArray array helpers.
+   * If the context exists and it's been selected, remove from array.
+   * Else push to the Formik FieldArray array.
+   */
+  const handleSelect = React.useCallback(
+    (arrayHelpers: FieldArrayRenderProps, contextName: string) => {
+      const currentContext: ContextOption = selectedContexts.find(
+        (ctx: ContextOption) => ctx.name === contextName,
+      );
+      const isSelected = currentContext && currentContext.selected;
+      const index = contexts.findIndex((c: ContextOption) => c.name === contextName);
+
+      if (isSelected && index !== -1) {
+        arrayHelpers.remove(index); // Deselect
+      } else if (!isSelected) {
+        // Select, add necessary data
+        arrayHelpers.push({ name: contextName, description: currentContext.description });
+      }
+    },
+    [contexts, selectedContexts],
+  );
+
+  return (
+    <div>
```

Comment:
use FormGroup Component from Patternfly to set the label and fieldId.

Reply:
FormGroup added with label and field id: https://github.com/openshift/hac-dev/pull/997/files#diff-f46d00580c9b2d98a2c8088a4544bea1d84ff334dd4914835883c3ab97dfc358R154